fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rewrite/rules/InPredicateDedup.java

```diff
@@ -0,0 +1,51 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.rules.expression.rewrite.rules;
+
+import org.apache.doris.nereids.rules.expression.rewrite.AbstractExpressionRewriteRule;
+import org.apache.doris.nereids.rules.expression.rewrite.ExpressionRewriteContext;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.InPredicate;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Deduplicate InPredicate, For example:
+ * where A in (x, x) ==> where A in (x)
+ */
+public class InPredicateDedup extends AbstractExpressionRewriteRule {
+
+    public static InPredicateDedup INSTANCE = new InPredicateDedup();
+
+    @Override
+    public Expression visitInPredicate(InPredicate inPredicate, ExpressionRewriteContext context) {
+        Set<Expression> dedupExpr = new HashSet<>();
+        List<Expression> newOptions = new ArrayList<>();
+        for (Expression option : inPredicate.getOptions()) {
+            if (dedupExpr.contains(option)) {
+                continue;
+            }
+            dedupExpr.add(option);
+            newOptions.add(option);
+        }
+        return new InPredicate(inPredicate.getCompareExpr(), newOptions);
```

Comment:
if there is only one option, it would be better to convert to equalTo

Reply:
InPredicateToEqualToRule will do it